algorithms/C/tree/MaxElementOfTree.c

```diff
@@ -0,0 +1,119 @@
+// Wap to find the maximum(using recursion) element in a binary search tree.
+
+#include<stdio.h>
+#include<stdlib.h>
+
+struct Node{
+    int data;
+    struct Node *left;
+    struct Node *right;
+};
+
+// creating a Node
+struct Node * createNode(int data){
+	struct Node *n;
+    n = (struct Node *)malloc(sizeof(struct Node)); // allocating the memory in the heap
+    n -> data = data; // setting the data
+    n -> left = NULL; // setting left node to be NULL
+    n -> right = NULL; // setting right node to be NULL
+    return n; // returning the created node
+}
+
+//function to insert a node in tree
+struct Node *insert(struct Node *root, int data){
+	if(root==NULL){ // empty tree condition
+		root = createNode(data);
+		return root;
+	}
+	
+	// if tree is not empty (data to be inserted at left subtree)
+	// data is to be inserted and root->data is value of the root node to be compared with passed data
+	else if(data <= root->data)
+		root->left = insert(root->left,data);
+	
+	// if tree is not empty (data to be inserted at right subtree)
+	else if(data > root->data)
+		root->right = insert(root->right,data);
+	return root;
+}
+
+//function to find the min element using recursion
+int findMin(struct Node *root){
+	if(root==NULL){
+		printf("Tree is empty\n");
+		return -1;

```

Comment:
Why you add this function here there is no need to find Min Value in tree because this program main motive is to find maximum value in the tree